spring-batch-excel/src/main/java/org/springframework/batch/item/excel/poi/PoiSheet.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2006-2014 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.batch.item.excel.poi;
+
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.FormulaEvaluator;
+import org.apache.poi.ss.usermodel.Row;
+import org.springframework.batch.item.excel.Sheet;
+
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Sheet implementation for Apache POI.
+ *
+ * @author Marten Deinum
+ * @since 0.5.0
+ */
+public class PoiSheet implements Sheet {
+
+    private final org.apache.poi.ss.usermodel.Sheet delegate;
+
+    private int numberOfColumns = -1;
+
+    /**
+     * Constructor which takes the delegate sheet.
+     *
+     * @param delegate the apache POI sheet
+     */
+    PoiSheet(final org.apache.poi.ss.usermodel.Sheet delegate) {
+        super();
+        this.delegate = delegate;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int getNumberOfRows() {
+        return this.delegate.getLastRowNum() + 1;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getName() {
+        return this.delegate.getSheetName();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String[] getRow(final int rowNumber) {
+        final Row row = this.delegate.getRow(rowNumber);
+        if (row == null) {
+            return null;
+        }
+        final List<String> cells = new LinkedList<String>();
+
+        for (int i = 0; i < getNumberOfColumns(); i++) {
+            Cell cell = row.getCell(i);
+            switch (cell.getCellType()) {
+                case Cell.CELL_TYPE_NUMERIC:
+                    cells.add(String.valueOf(cell.getNumericCellValue()));
```

Comment:
Been checking this extension out for use in a project and was just wondering if there was a reason the `CELL_TYPE_NUMERIC` case does not check for a Date formatted numeric as shown in the POI quick guide (see http://poi.apache.org/spreadsheet/quick-guide.html#CellContents). Without this the date is returned as the serialized number. Wasn't sure if this was the intention so thought I'd mention it.

From POI guide:

```
case Cell.CELL_TYPE_NUMERIC:
    if (DateUtil.isCellDateFormatted(cell)) {
        System.out.println(cell.getDateCellValue());
    } else {
        System.out.println(cell.getNumericCellValue());
    }
    break;
```

My apologies if this was the incorrect place to note this.